infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -0,0 +1,78 @@
+package org.corfudb.infrastructure.orchestrator;
+
+import io.netty.channel.ChannelHandlerContext;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.IServerRouter;
+import org.corfudb.infrastructure.orchestrator.workflows.AddNode;
+import org.corfudb.protocols.wireprotocol.CorfuMsgType;
+import org.corfudb.protocols.wireprotocol.CorfuPayloadMsg;
+import org.corfudb.protocols.wireprotocol.orchestrator.AddNodeRequest;
+import org.corfudb.protocols.wireprotocol.orchestrator.OrchestratorRequest;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Orchestrator is a service that acts on executing actions that belong to workflow instances.
+ * <p>
+ * Created by Maithem on 10/25/17.
+ */
+
+@Slf4j
+public class Orchestrator {
+
+    final CorfuRuntime runtime;
+
+    /**
+     * The id of this management server.
+     */
+    final UUID id;
+
+    public Orchestrator(CorfuRuntime runtime) {
+        this.runtime = runtime;
+        this.id = UUID.randomUUID();
+    }
+
+    public void handle(CorfuPayloadMsg<OrchestratorRequest> msg,
+                       ChannelHandlerContext ctx, IServerRouter r) {
+
+        OrchestratorRequest req = msg.getPayload();
+        dispatch(req);
+        r.sendResponse(ctx, msg, CorfuMsgType.ORCHESTRATOR_RESPONSE.msg());
+    }
+
+    void dispatch(OrchestratorRequest entry) {
+        CompletableFuture.runAsync(() -> run(entry));
+    }
+
+    private Workflow getWorkflow(OrchestratorRequest req) {
+        switch (req.getWorkFlowType()) {
+            case ADD_NODE:
+                AddNodeRequest addNodeRequest = (AddNodeRequest) req;
+                return new AddNode(addNodeRequest.getEndpoint());
+            default:
+                throw new RuntimeException("Unknown workflow");
+        }
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)